src/pipelines/worldpop/worldpop_population.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import zipfile
+from io import BytesIO
+from functools import partial
+from typing import Any, Dict, List, Tuple
+
+import requests
+from tqdm import tqdm
+from pandas import DataFrame, Series, read_csv, isnull
+
+from lib.concurrent import thread_map
+from lib.net import download
+from lib.pipeline import DataSource
+from lib.cast import age_group
+
+
+class WorldPopPopulationDataSource(DataSource):
+    """ Retrieves the requested properties from Wikidata for all items in metadata.csv """
+
+    url_strat_tpl = (
+        "ftp://ftp.worldpop.org.uk/GIS"
+        "/AgeSex_structures/Global_2000_2020/2020/"
+        "{code}/{code_lower}_{sex}_{age_bucket}_2020.tif"
+    )
+
+    url_total_tpl = (
+        "ftp://ftp.worldpop.org.uk/GIS"
+        "/Population/Global_2000_2020/2020/{code}/{code_lower}_ppp_2020.tif"
+    )
+
+    @staticmethod
+    def aggregate_population(data: Series) -> Dict[str, int]:
```

Comment:
Is there any reason this needs to be a public static function? Since this only appears to be called as a helper function in the `parse` function below. Can we take this function out of the class and make it a module private function?

Reply:
I'm not sure what the best practice is here, but I agree with you. How would you recommend defining functions which are only used privately within the class? I normally do @staticmethod and use the "_" prefix (I forgot the prefix here)

Reply:
Just as a module-level function `_aggregate_population(...)` defined at module scope.

Reply:
OK, so outside of the class. That makes sense. Done.